Merge upstream OTP2

leonardehrenfried opened the pull request:
This contains the GTFS spec alignment PR contributed by myself.

That means that after the merge you don't need duplicate stop_time entries for flex routes any more if you want to get on and off at inside the same zone.

cc @philip-cline

richardkoszegi committed:
Read bike parking polygons

richardkoszegi committed:
Change BikePark to VehicleParking

richardkoszegi committed:
Replace ParkAndRide classes with VehicleParking

richardkoszegi committed:
Add FeedId parameter to bicycleUpdater config

richardkoszegi committed:
Read capacity fields from OSM

richardkoszegi committed:
Extract VehicleParkingService from BikeRentalStationService

richardkoszegi committed:
Rename bike park updater classes to vehicle parking

richardkoszegi committed:
Merge bikeParkAndRide into parkAndRide in RoutingRequest

... and add carParkCost, carParkTime

richardkoszegi committed:
Handling vehicle parking entrances

richardkoszegi committed:
Merge P+R and B+R OSM processing

flaktack committed:
Connect the inner holes of multipolygons to the street network

flaktack committed:
Add details of the used VehicleParking to the REST API

flaktack committed:
Create a new leg when parking a bike / car

Since the the parking are may be entered through different entrances,
the leg.to / leg.from may be different places. The time for parking is
added to the walking leg.

richardkoszegi committed:
I18NString hashCode implementations

flaktack committed:
Add bannedVehicleParkingTags and requiredVehicleParkingTags

... to limit the used vehicle parkings.

richardkoszegi committed:
Allow setting the VehicleParking entrance name using the access nodes

richardkoszegi committed:
Correct check for P+R elements

... so that nodes with only a "park_ride != no" tag are not classified
as car park and rides.

This requires connecting the test files since the amenity=parking tags
was missing from the park_ride areas.

richardkoszegi committed:
Update legacy GraphQL Schema with VehicleParking

flaktack committed:
SandboxAPITransmodelApi: restore existing functionality

flaktack committed:
SandboxAPIParkAndRideApi: only return VehicleParkings with support for cars

This restores the previous functionality of the API.

flaktack committed:
Use CreativeName generated names for VehicleParkings as fallbacks

This allows the default text to be translated and restores previous
functionality.

richardkoszegi committed:
LegacyGraphQL API: keep BikePark and CarPark object

dominikjambor committed:
Added tests for planning with wheelchair accessible park and ride

flaktack committed:
Update vehicle parking VertexType values

flaktack committed:
VehicleParking: update documentation and clarify naming

hannesj committed:
Check if vertices were correctly linked, and throw exception if not.

hannesj committed:
Change PlanResponse messages to be RoutingErrors

hannesj committed:
Add routing errors in post-processing, if no itineraries are returned

hannesj committed:
Add routingErrors in the Transmodel GraphQL API

hannesj committed:
Distinguish whether the coordinates are inside the routing polygon

hannesj committed:
Add changelog entry

hannesj committed:
Add only a single RoutingError per type

This happens, when both the direct search and transit access/egress search adds routing errors

hannesj committed:
Properly set error message in REST API

Fixes #3581

hannesj committed:
Review issues

hannesj committed:
Apply suggestions from code review

Co-authored-by: Thomas Gran <t2gran@gmail.com>

hannesj committed:
Merge remote-tracking branch 'origin/dev-2.x' into otp2_improve_errors

hannesj committed:
Do not mix input and output arguments

hannesj committed:
Extract common logic

hannesj committed:
Merge remote-tracking branch 'origin/dev-2.x' into otp2_improve_errors

hannesj committed:
Merge branch 'dev-2.x' into otp2_improve_errors

hannesj committed:
Merge branch 'dev-2.x' into otp2_improve_errors

t2gran committed:
Add cost to maximize wait-time and avoid back-travel

hannesj committed:
Update variable name

hannesj committed:
Always add plan to Transmodel plan API response

hannesj committed:
Fix error in getting resource bundle

Locale is not injected in Transmodel API

flaktack committed:
Make VehicleParking available from it's entrances

src/main/java/org/opentripplanner/graph_builder/module/osm/OpenStreetMapModule.java
src/main/java/org/opentripplanner/routing/vehicle_parking/VehicleParking.java

flaktack committed:
Add allowedBikeRentalNetworks and bannedBikeRentalNetworks

... so that the networks used during planning may be restricted.

hannesj committed:
Remove changelog entry

Co-authored-by: Thomas Gran <t2gran@gmail.com>

flaktack committed:
Remove lombok annotations only leaving RoutingService.

A few typos were also fixed along the way.

leonardehrenfried committed:
Update documentation and migrate test to Junit 5

leonardehrenfried committed:
Remove unused code

flaktack committed:
Add TODO to ApiRouterInfo to use a mapper

Co-authored-by: Thomas Gran <t2gran@gmail.com>

flaktack committed:
Merge remote-tracking branch 'upstream/dev-2.x' into feature/vehicle-parking

flaktack committed:
VehicleParking: use request locale for mapping entrance names

hannesj committed:
Add failing test

hannesj committed:
Support correct types for transfer after flex access

leonardehrenfried committed:
Allow flex trips with a single stop time

flaktack committed:
Add documentation to VehicleParking fields

flaktack committed:
Rename VertexTypes and add backward-compatible mappings

flaktack committed:
LegacyGraphQLApi: add documentation

flaktack committed:
LegacyGraphQLApi: correct merge artifact

flaktack committed:
LegacyGraphQLApi: use backward-compatible vertex types

PARKANDRIDE is not present in VertexType, and so it is not mapped to the
GraphQL type.

flaktack committed:
LegacyGraphQLApi: preserve backward-compatability in type resolvers

leonardehrenfried committed:
Allow UnscheduledTrip instances with a single stop_time

leonardehrenfried committed:
Split large test into separate ones

leonardehrenfried committed:
Fix links to ActuatorAPI.md

flaktack committed:
LegacyGraphQLApi: update documentation

leonardehrenfried committed:
Add changelog entry

leonardehrenfried committed:
Only run automatic changelog in the upstream repository

flaktack committed:
LegacyGraphQLApi: deprecate BikePark/CarPark

leonardehrenfried committed:
Add test for Aspen flex feed

leonardehrenfried committed:
Take newly introduced flex params into account

leonardehrenfried committed:
Rename 'Health API' to 'Actuator API'

leonardehrenfried committed:
Rename token name to CHANGELOG_TOKEN and add more if clauses

leonardehrenfried committed:
Extract reusable test logic

leonardehrenfried committed:
Allow getting on and off at the same stop location for ScheduledDeviatedTrip

t2gran committed:
Merge pull request #3722 from ibi-group/changelog-limit

Only run automatic changelog in the upstream repository [changelog skip]

t2gran committed:
fix quotes in "automatic-changelog.yml" script

leonardehrenfried committed:
Improve changelog entries

hannesj committed:
Add todo

t2gran committed:
refactor: Do not print "cent" for cost above 10_000.

hannesj committed:
Add MicrometerGraphQLInstrumentation

hannesj committed:
Simplify logic

hannesj committed:
Merge remote-tracking branch 'origin/dev-2.x' into otp2_improve_errors

hannesj committed:
Include errors, even if a direct street search returns a result

hannesj committed:
Update documentation

hannesj committed:
Review comments

flaktack committed:
Merge remote-tracking branch 'upstream/dev-2.x' into feature/vehicle-parking

hannesj committed:
fix formatting strings

flaktack committed:
VehicleParking: avoid double linking

hannesj committed:
Merge pull request #3729 from entur/otp2_fix_string_templating

Fix string formatting in NeTEx import issue report [changelog skip]

hannesj committed:
Merge pull request #3727 from entur/otp2_graphql_micrometer

Add Micrometer monitoring for GraphQL [changelog skip]

leonardehrenfried committed:
Correct spelling

flaktack committed:
LegacyGraphQLApi: correctly deprecate fields

leonardehrenfried committed:
Merge pull request #3720 from ibi-group/flex-spec-alignment

Allow Flex trips to use the same stop_time for boarding and alighting [changelog skip]

leonardehrenfried committed:
Move fare services into their separate package

leonardehrenfried committed:
Fix build by using the correct imports

leonardehrenfried committed:
Merge pull request #3730 from ibi-group/fix-build

Fix build by using the correct imports [changelog skip]

hannesj committed:
Merge pull request #3628 from entur/otp2_improve_errors

Add new routing errors for cases where all itineraries were filtered by post-processing

Commit:
Add changelog entry for #3628 [ci skip]

flaktack committed:
LegacyGraphQLApi: use backward compatible bikePark/carPark IDs

miles-grant-ibigroup reviewed (approved):
Thanks for making the PR! Looks great

hannesj committed:
Update src/test/java/org/opentripplanner/transit/raptor/moduletests/E11_GuaranteedTransferWithFlexAccessTest.java

Co-authored-by: Leonard Ehrenfried <mail@leonard.io>

t2gran committed:
Merge pull request #3726 from entur/otp2_fix_flex_transfers

Fix combination of flex access and constrained transfer

Commit:
Add changelog entry for #3726 [ci skip]

flaktack committed:
LegacyGraphQLApi: use backward compatible IDs in carParks

leonardehrenfried committed:
Make FareService use Itinerary instead of RaptorPath

leonardehrenfried committed:
Remove unused TransitLayer argument

t2gran committed:
Review: revert doc changes.

leonardehrenfried committed:
Properly check for a transfer leg in NYC

leonardehrenfried committed:
Refactor multiple fares test

leonardehrenfried committed:
Use correct time calculation

flaktack committed:
Merge pull request #3480 from realCity/feature/vehicle-parking

Merge B+R and P+R functionality into vehicle parking

Commit:
Add changelog entry for #3480 [ci skip]

leonardehrenfried committed:
Fix the calculation of the zones that the ride passes through

hannesj committed:
fix: Ignore flexibleOnly default value

hannesj committed:
fix: do not throw NPE in Place constructor

leonardehrenfried committed:
Set intermediateStops to null if not requested

leonardehrenfried committed:
Remove empty intermediateStops

lassetyr committed:
Logging first time instance reports it is ready to use

leonardehrenfried committed:
Convert to LocalTime for easier handling

leonardehrenfried committed:
Use reusable method to fix the build

t2gran committed:
feat: Remove AddMinSafeTransferCostFilter from itinerary-filter-chain

The AddMinSafeTransferCostFilter sometimes select itineraries with a lot of walking
instead of itineraries with short high frequency transit legs when there is a tight
connections. Overriding the Raptor router this way is not correct, and the best way
to do this is to improve this in the routing itself.

leonardehrenfried committed:
Merge pull request #3735 from ibi-group/fix-build

Use reusable method to fix the build [changelog skip]

t2gran committed:
Update src/ext/java/org/opentripplanner/ext/transmodelapi/model/plan/ItineraryFiltersInputType.java

Co-authored-by: Hannes Junnila <hannes.junnila@gmail.com>

t2gran committed:
Merge pull request #3738 from entur/otp2_fix_transmodel_ids_filter

Transmodel API: Allow filtering by a list of ids [changelog skip]

hannesj committed:
Merge pull request #3737 from entur/otp2_fix_place_npe

Do not throw NPE in Place constructor [changelog skip]

leonardehrenfried committed:
Start revamping tests

flaktack committed:
Add B+R/P+R cost / time parameters to RoutingResource

leonardehrenfried committed:
Improve test setup

leonardehrenfried committed:
Improve reusability of graph generation

leonardehrenfried committed:
Reactivate another test

leonardehrenfried committed:
Fix KCM test

leonardehrenfried committed:
Improve robustness of flex tests

leonardehrenfried committed:
Remove test that is based on frequencies

t2gran committed:
refactor: Remove dependency from updaters to GraphUpdaterManager

A few code cleanup is applied as well.

t2gran committed:
feat: Log all updaters initialized

t2gran committed:
Merge pull request #3736 from entur/otp2_remove_add-min-safe-transfer-cost-filter

Remove AddMinSafeTransferCostFilter from itinerary-filter-chain

Commit:
Add changelog entry for #3736 [ci skip]

leonardehrenfried committed:
Make rides conversion more robust

leonardehrenfried committed:
Convert from LocalTime to ZonedDateTime

t2gran committed:
feat: Move the numberOfNonePrimedUpdaters updates into a designated thread.

leonardehrenfried committed:
Move test files into subfolder

leonardehrenfried committed:
Reactivate fare component tests

leonardehrenfried committed:
Take random stop order into account

leonardehrenfried committed:
Calculate fares for direct flex routes

leonardehrenfried committed:
Add changelog

t2gran committed:
Merge pull request #3654 from entur/otp2_fixoptimize_transfers

Add cost to maximize wait-time and avoid back-travel in optimize transfers

Commit:
Add changelog entry for #3654 [ci skip]

t2gran committed:
review: Make sure the name of waiting updaters are reported.

leonardehrenfried committed:
Add forked version of oba-gtfs

leonardehrenfried committed:
Merge pull request #3741 from ibi-group/fix-flex-trips-test

Fix flex tests by taking into account that stops are a Set [changelog skip]

leonardehrenfried committed:
Adjust test to new GTFS feed

leonardehrenfried committed:
Improve documentation

flaktack committed:
debug-client: split additional parameters while allowing for multiple = to be present

Vehicle parking tags may contain an (=) when specified for
requiredVehicleParkingTags / bannedVehicleParkingTags.

leonardehrenfried committed:
Update src/ext-test/java/org/opentripplanner/ext/flex/ScheduledDeviatedTripTest.java

Co-authored-by: Thomas Gran <t2gran@gmail.com>

leonardehrenfried committed:
Move mapping of intermediate stops to API mapper

leonardehrenfried committed:
Use released version of OBA

t2gran committed:
review: Change 'waitingForUpdaters' to 'listUnprimedUpdaters' and update doc.

flaktack committed:
Reuse existing asTextSet when mapping RoutingRequest

t2gran committed:
Update Developer-Guide

hannesj committed:
fix: don't filter out stops who don't have multimodal parents

leonardehrenfried committed:
Merge pull request #3743 from ibi-group/flex-fares

Calculate fares from itineraries not Raptor paths, calculate flex fares

Commit:
Add changelog entry for #3743 [ci skip]

hannesj committed:
Merge branch 'dev-2.x' into otp2_fix_nearest_multimodal

t2gran committed:
Merge pull request #3733 from entur/otp2_log_readiness

Logging first time instance reports it is ready to use

Commit:
Add changelog entry for #3733 [ci skip]

flaktack committed:
Merge pull request #3746 from realCity/feature/vehicle-parking-filtering

Allow limiting the used vehicle rentals and parkings

Commit:
Add changelog entry for #3746 [ci skip]

t2gran committed:
Merge pull request #3752 from entur/otp2_fix_nearest_multimodal

Don't filter out stops who don't have multimodal parents in Transmodel API nearest query [changelog skip]

binh-dam-ibigroup reviewed (approved):
Let's go ahead with the upstream changes.
